src/main/java/at/ac/tuwien/kr/alpha/common/depgraph/DependencyGraphUtils.java

```diff
@@ -0,0 +1,147 @@
+	 * dfsDiscoveryTime, dfsFinishTime and dfsPredecessor values for each node are set. The algorithm follows the approach outlined in "Introduction to
+	 * Algortihms, 3rd. Edition" by Cormen et al. Note that no separate data structure for the discovered depth-first forest is returned as that information can
+	 * be gained from the completely filled <code>NodeInfo</code>s
+	 * 
+	 * @param nodeVisitSeq an Iterator defining in which sequence nodes should be visited
+	 * @param nodes        an adjacency map defining the dependency graph of an ASP program
+	 * @return a Set<Node> holding all finished nodes (i.e. all nodes at the end of the DFS run)
+	 */
+	public static DfsResult performDfs(Iterator<Node> nodeVisitIt, Map<Node, List<Edge>> nodes) {
+		DfsResult retVal = new DfsResult();
+		Set<Node> discovered = new HashSet<>();
+		Deque<Node> finished = new LinkedList<>();
+		Map<Node, List<Node>> depthFirstForest = new HashMap<>();
+		depthFirstForest.put(null, new ArrayList<>());
+		int dfsTime = 0;
+		Node tmp;
+		while (nodeVisitIt.hasNext()) {
+			tmp = nodeVisitIt.next();
+			if (!(discovered.contains(tmp) || finished.contains(tmp))) {
+				depthFirstForest.get(null).add(tmp);
+				dfsTime = DependencyGraphUtils.dfsVisit(dfsTime, tmp, nodes, discovered, finished, depthFirstForest);
+			}
+		}
+		retVal.setFinishedNodes(finished);
+		retVal.setDepthFirstForest(depthFirstForest);
+		return retVal;
+	}
+
+	private static int dfsVisit(int dfsTime, Node currNode, Map<Node, List<Edge>> nodes, Set<Node> discovered, Deque<Node> finished,
+			Map<Node, List<Node>> dfForest) {
+		int retVal = dfsTime;
+		retVal++;
+		currNode.getNodeInfo().setDfsDiscoveryTime(retVal);
+		discovered.add(currNode);
+		Node tmpNeighbor;
+		for (Edge e : nodes.get(currNode)) {
+			// progress to adjacent nodes
+			tmpNeighbor = e.getTarget();
+			if (!(discovered.contains(tmpNeighbor) || finished.contains(tmpNeighbor))) {
+				tmpNeighbor.getNodeInfo().setDfsPredecessor(currNode);
+				if (!dfForest.containsKey(currNode)) {
+					dfForest.put(currNode, new ArrayList<>());
+				}
+				dfForest.get(currNode).add(tmpNeighbor);
+				retVal = DependencyGraphUtils.dfsVisit(retVal, tmpNeighbor, nodes, discovered, finished, dfForest);
+			}
+		}
+		retVal++;
+		currNode.getNodeInfo().setDfsFinishTime(retVal);
+		finished.add(currNode);
+		return retVal;
+	}
+
+	public static boolean isReachableFrom(Node dest, Node src, DependencyGraph dg) {
+		return DependencyGraphUtils.isReachableFrom(dest, src, dg, new ArrayList<>());
+	}
+
+	private static boolean isReachableFrom(Node dest, Node src, DependencyGraph dg, List<Node> discovered) {
+		List<Edge> outgoingEdges;
+		if (src.equals(dest)) {
+			return true;
+		}
+		if ((outgoingEdges = dg.getNodes().get(src)) == null) {
+			return false;
+		}
+		discovered.add(src);
+		// we wanna do BFS here, therefore use 2 loops
```

Comment:
This could need some better explanation; also please use full sentences for comments.